Sources/Workspace/Workspace+State.swift

```diff
@@ -254,7 +254,7 @@ extension WorkspaceStateStorage {
                         let version = try container.decode(String.self, forKey: .version)
                         return try self.init(underlying: .registryDownload(version: TSCUtility.Version(versionString: version)))
                     case "edited":
-                        let path = try container.decode(AbsolutePath?.self, forKey: .path)
+                        let path = try container.decode(AbsolutePath.self, forKey: .path)
```

Comment:
Are these changes related?

Reply:
Unrelated but required for this package to build in the first place with a tip-of-tree (or near-to-that) swiftc; see PR descriptions: 
> unrelated fixes to appease snapshot swiftc where AbsolutePath?.self is no longer valid Swift.

Reply:
I dont think we have any issue building SwiftPM with ToT. where have you seen this to be an issue?

cc @neonichu

Reply:
Sorry, I didn't note down the failure when I saw it, and now I can't reproduce it anymore. 

I'm fine reverting this aspect of the PR, but is there a context in which the `?` is adding value?

Reply:
it is saying that the value is optional so if the serialized formal (in this case I think it is JSON) does not have it its fine. this is important for backwards compatibility

Reply:
Oh, I see. Thanks for catching this. Reverted.